modules/core/src/main/scala/org/tessellation/l0/http/Routes.scala

```diff
@@ -0,0 +1,29 @@
+package org.tessellation.l0.http
+import cats.effect.Async
+import cats.effect.std.Queue
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import org.tessellation.dag.domain.block.DAGBlock
+import org.tessellation.l0.domain.snapshot.SnapshotService
+import org.tessellation.security.signature.Signed
+
+import org.http4s.HttpRoutes
+import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
+import org.http4s.dsl.Http4sDsl
+
+final case class Routes[F[_]: Async](
+  l1BlocksQueue: Queue[F, Signed[DAGBlock]],
+  snapshotService: SnapshotService[F]
+) extends Http4sDsl[F] {
+
+  private val p2p: HttpRoutes[F] = HttpRoutes.of[F] {
+    case req @ POST -> Root / "block" =>
+      for {
+        block <- req.as[Signed[DAGBlock]]
+        response <- snapshotService.offer(block)
+        res <- Ok()
+      } yield res
```

Comment:
IMO more clear is
```scala
req.as[Signed[DAGBlock]] >>
  snapshotService.offer >>
  Ok()
```
or 
```scala
req.as[Signed[DAGBlock]]
  .flatMap(snapshotService.offer)
  .flatMap(_ => Ok())
```

It's easy to do sth like that then:
```scala
req.as[Signed[DAGBlock]]
  .flatMap(snapshotService.offer)
  .flatMap(_ => Ok())
  .recoverWith {
    case MyError => Conflict(...)
    case MyOtherError => NotFound(...)
  }
```

Reply:
That code will be modified by the implementation of tip-based trigger. But I agree :)